ALUM/ALUM/Models/CurrentUserModel.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  CurrentUserModel.swift
+//  ALUM
+//
+//  Created by Yash Ravipati on 5/4/23.
+//
+
+import SwiftUI
+import FirebaseAuth
+
+enum UserRole {
+    case mentor
+    case mentee
+}
+
+class CurrentUserModal: ObservableObject {
+    static let shared = CurrentUserModal()
+
+    @Published var isLoading: Bool
+    @Published var uid: String?
+    @Published var role: UserRole?
+    @Published var isLoggedIn: Bool
+
+    init() {
+        self.isLoading = true
+        self.isLoggedIn = false
+        self.uid = nil
+        self.role = nil
+    }
+
+    ///  Since async operations are involved, this function will limit updating the current
+    ///  user without using the DispatchQueue logic.
+    ///  Not using DispatchQueue can casue race conditions which can crash our app
+    func setCurrentUser(isLoading: Bool, isLoggedIn: Bool, uid: String?, role: UserRole?) {
+        DispatchQueue.main.async {
```

Comment:
important - we should use this while updating @published states
https://sharegpt.com/c/GfFG34x